src/gui/sharepermissions.h

```diff
@@ -28,7 +28,7 @@ enum SharePermission {
     SharePermissionCreate = 4,
     SharePermissionDelete = 8,
     SharePermissionShare = 16,
-    SharePermissionDefault = 1 << 30
+    SharePermissionDefault = 31
```

Comment:
I wonder, isn't it badly named now? From your description the default can be 31 or 1 depending if it's a public share or not.

I don't know that area well though, so it might just be me being confused.